packages/query-db-collection/src/query.ts

```diff
@@ -304,6 +305,119 @@ class QueryCollectionUtilsImpl {
   }
 }
 
+/**
+ * Serializes LoadSubsetOptions into a stable, hashable format for query keys
+ * @internal
+ */
+function serializeLoadSubsetOptions(
+  options: LoadSubsetOptions | undefined
+): unknown {
+  if (!options) {
+    return null
+  }
+
+  const result: Record<string, unknown> = {}
+
+  if (options.where) {
+    result.where = serializeExpression(options.where)
+  }
+
+  if (options.orderBy?.length) {
+    result.orderBy = options.orderBy.map((clause) => ({
+      expression: serializeExpression(clause.expression),
+      direction: clause.compareOptions.direction,
+      nulls: clause.compareOptions.nulls,
```

Comment:
This object only contains the `direction` and `nulls` but is missing the other options from `StringCollationConfig`:
```ts
export type CompareOptions = StringCollationConfig & {
  direction: OrderByDirection
  nulls: `first` | `last`
}
```